src/Java.Interop/Java.Interop/JniRuntime.cs

```diff
@@ -414,16 +414,12 @@ partial class JniRuntime {
 
 		public virtual void RaisePendingException (Exception pendingException)
 		{
+#if XA_INTEGRATION
 			if (pendingException == null)
 				throw new ArgumentNullException (nameof (pendingException));
-#if XA_INTEGRATION
 			throw new NotSupportedException ("Do not know how to marshal System.Exception instances.");
-#else   // XA_INTEGRATION
-			var je  = pendingException as JavaException;
```

Comment:
I don't understand why any of this code is being removed.

Reply:
The same code is in https://github.com/radekdoulik/java.interop/blob/b9ee7b1527fee332d9e1323790a7c6847984a3c2/src/Java.Interop/Java.Interop/JniEnvironment.Errors.cs#L32 which we call below.